p4-16/spec/grammar.mdk

```diff
@@ -57,6 +57,7 @@ nonEmptyParameterList
 
 parameter
     : optAnnotations direction typeRef name
+    | optAnnotations direction typeRef name '=' expression
```

Comment:
The grammar seems to imply `@optional in bit<16> x = 0` is a valid syntax. But it is actually not according to the current spec.

Reply:
Watch out: optAnnotation does not mean "the annotation `@optional`", but it means "with some optional annotations", which could be e.g., `@hidden`.

Reply:
Yes, I agree, but if `optional` is a keyword, then we won't have this ambiguity.

Reply:
At this point optional is not a keyword. We could attempt to promote it into a keyword. Maybe it is a good idea.

Reply:
Since it is an annotation, it seems safe to consider it not a keyword for now.